runtime/near-vm-logic/src/alt_bn128.rs

```diff
@@ -253,6 +246,11 @@ pub fn alt_bn128_g1_multiexp(data: &[u8]) -> crate::logic::Result<Vec<u8>> {
         .into_iter()
         .map(|e| (e.0 .0, e.1 .0))
         .collect::<Vec<_>>();
+    // Upper bounded by 2^9 buckets
+    // Reference: https://github.com/zeropoolnetwork/bn/blob/b25e138cd5a7d98ad16c091c20b4fa273ca1f993/src/groups/mod.rs#L71
+    if items.len() > 5000 {
+        return Err(VMLogicError::HostError(HostError::AltBn128MaxNumberOfItemsExceeded));
+    }
```

Comment:
the bn library used can cause an OOM error because it allocates buckets exponentailly, which is why this limit was added to circumvent that. The other solution would be to fork the library and add the constraint there

Reply:
https://github.com/zeropoolnetwork/bn/blob/b25e138cd5a7d98ad16c091c20b4fa273ca1f993/src/groups/mod.rs#L71

Reply:
I don't understand this reasoning. It seems that we allocate (`2^ln(items.len())`) buckets, where `ln` is a natural (base e) logarithm. That's going to be smaller than `items.len()`.

Reply:
Yeah, manual experiments didn't show exponential memory usage here.